pkg/build/fuchsia.go

```diff
@@ -52,22 +52,34 @@ func (fu fuchsia) build(params *Params) error {
 		return err
 	}
 
-	// Fuchsia images no longer include ssh keys. Manually append the ssh public key to the zbi.
-	sshZBI := filepath.Join(params.KernelDir, "out", arch, "fuchsia-ssh.zbi")
+	_, sshKeyPub, err := genSSHKeys(params.OutputDir)
+	if err != nil {
+		return err
+	}
+
+	sshZBI := filepath.Join(params.OutputDir, "initrd")
 	kernelZBI := filepath.Join(params.KernelDir, "out", arch, "fuchsia.zbi")
-	authorizedKeys := fmt.Sprintf("data/ssh/authorized_keys=%s",
-		filepath.Join(params.KernelDir, ".ssh", "authorized_keys"))
-	if _, err := runSandboxed(time.Minute, params.KernelDir, "out/"+arch+".zircon/tools/zbi",
+	authorizedKeys := fmt.Sprintf("data/ssh/authorized_keys=%s", sshKeyPub)
+
+	if _, err := osutil.RunCmd(time.Minute, params.KernelDir, "out/"+arch+"/host_x64/zbi",
 		"-o", sshZBI, kernelZBI, "--entry", authorizedKeys); err != nil {
 		return err
 	}
 
+	// Copy and extend the fvm.
+	fvmTool := filepath.Join("out", arch, "host_x64", "fvm")
+	fvmDst := filepath.Join(params.OutputDir, "image")
+	fvmSrc := filepath.Join(params.KernelDir, "out", arch, "obj/build/images/fvm.blk")
+	if err := osutil.CopyFile(fvmSrc, fvmDst); err != nil {
+		return err
+	}
+	if _, err := osutil.RunCmd(time.Minute*5, params.KernelDir, fvmTool, fvmDst, "extend", "--length", "3G"); err != nil {
+		return err
+	}
+
 	for src, dst := range map[string]string{
```

Comment:
Nit: At this point, might be simpler to just make this two copies instead of a loop?

Reply:
Let's keep it like that, we can refactor that in the future.